code/lib/manager-api/src/modules/layout.ts

```diff
@@ -284,7 +389,6 @@ export const init: ModuleFn = ({ store, provider, singleStory, fullAPI }) => {
     api,
     state: merge(api.getInitialOptions(), persisted),
     init: () => {
-      api.setOptions(merge(api.getInitialOptions(), persisted));
```

Comment:
this makes setting layout from URL work again, similar to https://github.com/storybookjs/storybook/pull/23750